autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/FastICA.py

```diff
@@ -0,0 +1,102 @@
+from math import ceil, floor
+from typing import Any, Dict, List, Optional
+
+from ConfigSpace.conditions import EqualsCondition
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import (
+    CategoricalHyperparameter,
+    UniformIntegerHyperparameter,
+)
+
+import numpy as np
+
+from sklearn.base import BaseEstimator
+from sklearn.decomposition import FastICA as SklearnFastICA
+
+from autoPyTorch.datasets.base_dataset import BaseDatasetPropertiesType
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.feature_preprocessing. \
+    base_feature_preprocessor import autoPyTorchFeaturePreprocessingComponent
+from autoPyTorch.utils.common import HyperparameterSearchSpace, add_hyperparameter, get_hyperparameter
+
+
+class FastICA(autoPyTorchFeaturePreprocessingComponent):
+    def __init__(self, n_components: Optional[int] = None,
+                 algorithm: str = 'euclidean', whiten: str = 'ward',
+                 fun: str = "max",
+                 random_state: Optional[np.random.RandomState] = None
+                 ):
+        self.n_components = n_components
+        self.algorithm = algorithm
+        self.whiten = whiten
+        self.fun = fun
+
+        super().__init__(random_state=random_state)
+
+    def fit(self, X: Dict[str, Any], y: Any = None) -> BaseEstimator:
+
+        self.preprocessor['numerical'] = SklearnFastICA(
+            n_components=self.n_components, algorithm=self.algorithm,
+            fun=self.fun, whiten=self.whiten, random_state=self.random_state)
+
+        return self
+
+    @staticmethod
+    def get_hyperparameter_search_space(
+        dataset_properties: Optional[Dict[str, BaseDatasetPropertiesType]] = None,
+        n_components: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='n_components',
+                                                                            value_range=(0.5, 0.9),
+                                                                            default_value=0.5,
+                                                                            ),
```

Comment:
The name is confusing?
In the doc, it is integer, so should we rephrase it?

Reply:
I think it's fine, we can explain this in the documentation. But the idea behind keeping the same name is that it will be easier for the user to pass a search space update for that hyperparameter as it is of the same name passed to the init function as well which is uniform throughout the library.